lib/ruby_lsp/response_builders/completion.rb

```diff
@@ -0,0 +1,28 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
```

Comment:
We don't need to create these specific response builders for the requests that just need an array. We can instead specify the type parameter when instantiating to get the type safety.

That will allow us to get rid of most of these builder classes and then we only need the ones that truly have different behaviour than the base one.

Here's an [example](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20ResponseBuilder%0A%20%20extend%20T%3A%3ASig%0A%20%20extend%20T%3A%3AHelpers%0A%20%20extend%20T%3A%3AGeneric%0A%0A%20%20Type%20%3D%20type_member%20%7B%7B%20upper%3A%20Object%20%7D%7D%0A%0A%20%20def%20initialize%0A%20%20%20%20%40items%20%3D%20T.let%28%5B%5D%2C%20T%3A%3AArray%5BType%5D%29%0A%20%20end%0A%0A%20%20sig%20%7B%20params%28item%3A%20Type%29.void%20%7D%0A%20%20def%20%3C%3C%28item%29%0A%20%20%20%20%40items%20%3C%3C%20item%0A%20%20end%0Aend%0A%0Ainteger_builder%20%3D%20ResponseBuilder%5BInteger%5D.new%0A%23%20Can't%20push%20incorrect%20types%20to%20it%0Ainteger_builder%20%3C%3C%20%22hello!%22%0Ainteger_builder%20%3C%3C%20123).